webapp-django/crashstats/crashstats/views.py

```diff
@@ -1354,9 +1357,156 @@ def exploitable_crashes(
     ))
     context['version'] = versions
     context['report'] = 'exploitable'
+    context['new_link_url'] = reverse('crashstats:exploitability_report')
+    query_string = {'product': product}
+    if versions:
+        query_string['version'] = versions
+    context['new_link_url'] += '?' + urllib.urlencode(query_string, True)
     return render(request, 'crashstats/exploitability.html', context)
 
 
+@utils.json_view
+@pass_default_context
+@permission_required('crashstats.view_exploitability')
+def exploitability_report(request, default_context=None):
+    context = default_context or {}
+
+    if not request.GET.get('product'):
+        url = reverse('crashstats:exploitability_report')
+        url += '?' + urllib.urlencode({
+            'product': settings.DEFAULT_PRODUCT
+        })
+        return redirect(url)
+
+    available_products = {}
+    for product, versions in context['releases'].items():
+        if product not in available_products:
+            available_products[product] = []
+        for version_info in versions:
+            available_products[product].append(version_info['version'])
+
+    form = forms.ExploitabilityReportForm(
+        request.GET,
+        available_products=available_products,
+    )
+    if not form.is_valid():
+        return http.HttpResponseBadRequest(str(form.errors))
+
+    if request.GET.get('format') == 'json':
+        product = form.cleaned_data['product']
+        version = form.cleaned_data['version']
+
+        api = SuperSearchUnredacted()
+        # XXX ADRIAN, how do I use settings.EXPLOITABILITY_BATCH_SIZE
+        # to get a limited set?
+        params = {
+            'product': product,
+            'version': version,
+            '_results_number': 0,
+            '_aggs.signature': 'exploitability',
+        }
+        results = api.get(**params)
+        # For the current page, let's find out what all the signatures
+        # were and use that for a bug associations query later
+        try:
+            page = max(1, int(request.GET.get('page', 1)))
+        except ValueError:
+            page = 1
+
+        crashes = []
+        categories = ('high', 'none', 'low', 'medium', 'null')
+        for signature_facet in results['facets']['signature']:
+            # this 'signature_facet' will look something like this:
+            #
+            #  {
+            #      'count': 1234,
+            #      'term': 'My | Signature',
+            #      'facets': {
+            #          'exploitability': [
+            #              {'count': 1, 'term': 'high'},
+            #              {'count': 23, 'term': 'medium'},
+            #              {'count': 11, 'term': 'other'},
+            #
+            # And we only want to include those where:
+            #
+            #   null + none + low + medium + high > 4
```

Comment:
I do not understand this logic at all. What are you trying to achieve? What does that ``4`` value represent?